### ISSUE TITLE
PyPI release

### ISSUE THREAD

**Issue by apcamargo** [RasmussenLab/vamb#20]
Hi @jakobnissen 

I'm currently considering using Vamb in my work and I'd like to put it in Bioconda. Do you have any plans to upload Vamb to PyPI or should the Bioconda build download the source from this repository?

Thanks!

**Comment by jakobnissen** [comment]
Dear @apcamargo 

That's a great idea. I ❤️ Conda!
To be honest, I prefer having a single public "homepage" for Vamb here on GitHub, such that I do not have to think about keeping multiple public archives up-to-date. Thanks to the contribution of C. Titus Brown, installing Vamb from GitHub should not be too difficult using Pip.
So for the foreseeable future I will not be putting Vamb on PyPI, and it's better to build from source (preferrably the latest release and not `master`!).

Are there any advantages to having Vamb on PyPI I might have missed, other than it making the install process ever so slightly easier?

**Comment by apcamargo** [comment]
I don't think that there's any disadvantage in keeping the source only in GitHub aside from making the installation slightly easier, as you said.

So I'm going to write the recipe to build Vamb from the lastest release! Is there a new version on the horizon that I should wait for?

**Comment by jakobnissen** [comment]
No, there is no version upcoming. The last commits have been small and
almost cosmetic. 2.0.2 is what we're running with internally, just use that
:)

**Comment by ctb** [comment]
hi @jakobnissen I was eavesdropping and I wanted to drop in a few notes about pypi and other release strategies that we've found useful --

zeroth, big fan of conda :)

first, you can reserve the vamb name at pypi without putting anything there other than a short description and a pointer to github. That reserves the vamb name and makes sure other people don't step on it unintentionally. Worth doing as a one off if nothing else.

second, it's possible to construct a pip install string that will install directly from github! something like
```
pip install git+https://github.com/RasmussenLab/vamb.git@v2.0.1
```
should work. Could be useful for docs somewhere.

third, I wanted to suggest connecting vamb to zenodo, as per [this doc](https://guides.github.com/activities/citable-code/). It's a one time setup that should take about a minute, and it means that every time you tag a new release you get an archive of the source code with a DOI on zenodo. Free, in perpetuity, etc. This is one of my most frequent suggestions as a paper reviewer - to make your code releases archived and citable - because github itself doesn't serve as a code archive.

LMK if you want any help doing any of this, otherwise I'll go back into lurker mode :)

**Comment by jakobnissen** [comment]
Dear @apcamargo 

I've released a new version of Vamb, and I would like to update the conda recipe. It would be nice if I just did it - that way, I wouldn't have to bother you, and I get to learn something new.

I've seen [your commit in bioconda](https://github.com/RasmussenLab/vamb/releases/tag/3.0.0). Three questions
* Can I just copy your PR and change the relevant fields?
* What is the sha256 a hash of? The `.tar.gz`?
* I see you have changed the Python requirement from "python >=3.5" to "python". However, I have been assuming >3.5 and don't know if it works for versions below this. Is there any reason to not have >3.5?

**Comment by apcamargo** [comment]
Hey @jakobnissen!

- Bioconda now has a bot that detects new PyPI releases and creates automatics PRs. However it only changes the version number and the sha256. Any further changes need to be manually done. Here's the automatic PR for the new version of Vamb: https://github.com/bioconda/bioconda-recipes/pull/21357
- Yes. Bioconda builds the package from the source, so the sha256 is from the `tar.gz` file.
- Bioconda only builds for Python 3.6 and 3.7.

**Comment by jakobnissen** [comment]
@apcamargo Great, thank you! Looks like it's all set then.

**Comment by apcamargo** [comment]
The [CI is failing right now](https://circleci.com/gh/bioconda/bioconda-recipes/102519?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link). It seems Vamb can't find `kernel.npz`.

**Comment by jakobnissen** [comment]
Urgh, that'll teach me not to make a pre release and test it out properly :/ I've amended it in the latest release and will get it put on bioconda.

**Comment by apcamargo** [comment]
Thanks! I just merged the PR.
